NIO's additional targets not in docs

New targets were added to the SPI manifest: https://github.com/apple/swift-nio/pull/2329

but they don't appear in the docs yet:

https://swiftpackageindex.com/apple/swift-nio/main/documentation/NIOCore is a 404.

The [build](https://swiftpackageindex.com/builds/7F95B847-9465-4C87-98DF-AF1D5EBF35CD) is generating and uploading docs for all referenced targets:

```
========================================
GenerateDocs
========================================
Generating docs at path:  spi-builder-workspace/.docs/apple/swift-nio/main
Repository:               apple/swift-nio
Swift version used:       5.7
        .package(url: "https://github.com/apple/swift-docc-plugin", from: "1.0.0"),
Target:                   NIO
Building for debugging...
Build complete! (0.28s)
Generating documentation for 'NIO'...
Converting documentation...
Conversion complete! (0.28s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Target:                   NIOConcurrencyHelpers
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOConcurrencyHelpers'...
Converting documentation...
Conversion complete! (0.23s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
Target:                   NIOCore
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOCore'...
Converting documentation...
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/ByteBuffer-views.swift:164:35: warning: Topic reference 'ByteBuffer.reserveCapacity(_%3A)' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncAwaitSupport.swift:288:24: warning: Topic reference 'Swift/AsyncSequence' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncSequences/NIOThrowingAsyncSequenceProducer.swift:43:49: warning: Topic reference 'Swift/AsyncSequence' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift:82:16: warning: Topic reference 'Swift/AsyncSequence' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift:82:64: warning: Topic reference 'Swift/AsyncIterator' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncSequences/NIOThrowingAsyncSequenceProducer.swift:18:16: warning: Topic reference 'Swift/AsyncSequence' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncSequences/NIOThrowingAsyncSequenceProducer.swift:18:64: warning: Topic reference 'Swift/AsyncIterator' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift:106:49: warning: Topic reference 'Swift/AsyncSequence' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncAwaitSupport.swift:266:24: warning: Topic reference 'Swift/AsyncSequence' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncSequences/NIOAsyncSequenceProducer.swift:256:94: warning: Topic reference 'YieldResult' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncAwaitSupport.swift:288:24: warning: Topic reference 'Swift/AsyncSequence' couldn't be resolved. No local documentation matches this reference.
/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/Sources/NIOCore/AsyncAwaitSupport.swift:266:24: warning: Topic reference 'Swift/AsyncSequence' couldn't be resolved. No local documentation matches this reference.
Conversion complete! (0.98s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
Target:                   NIOEmbedded
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOEmbedded'...
Converting documentation...
Conversion complete! (0.45s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
Target:                   NIOFoundationCompat
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOFoundationCompat'...
Converting documentation...
Conversion complete! (0.10s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
Target:                   NIOHTTP1
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOHTTP1'...
Converting documentation...
Conversion complete! (0.22s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
Target:                   NIOPosix
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOPosix'...
Converting documentation...
Conversion complete! (0.18s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
Target:                   NIOTLS
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOTLS'...
Converting documentation...
Conversion complete! (0.07s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
Target:                   NIOWebSocket
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOWebSocket'...
Converting documentation...
Conversion complete! (0.13s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
Target:                   NIOTestUtils
Building for debugging...
Build complete! (0.17s)
Generating documentation for 'NIOTestUtils'...
Converting documentation...
Conversion complete! (0.10s)
Generated DocC archive at '/Users/builder/builds/o86TiJKT/3/finestructure/swiftpackageindex-builder/spi-builder-workspace/.docs/apple/swift-nio/main'
Successfully merged into spi-builder-workspace/.docs-staging/apple/swift-nio/main/index/index.json
========================================
SyncDocs
========================================
Syncing directory: spi-builder-workspace/.docs/apple/swift-nio/main
61	spi-builder-workspace/.docs/apple/swift-nio/main
Doc size: 61.0MB
S3 URL: s3://spi-prod-docs/apple/swift-nio/main
Sync docs (attempt 1)
Syncing... [10%]
Syncing... [20%]
Syncing... [30%]
Syncing... [40%]
Syncing... [50%]
Syncing... [60%]
Syncing... [70%]
Syncing... [80%]
Syncing... [90%]
Syncing... [100%]
Done.
```

Job logs: https://gitlab.com/finestructure/swiftpackageindex-builder/-/jobs/3418588819

--- a/Sources/App/Controllers/PackageController+routes.swift
+++ b/Sources/App/Controllers/PackageController+routes.swift
@@ -91,7 +91,7 @@ enum PackageController {
             throw Abort(.notFound)
         }
         let anchor = req.url.fragment.map { "#\($0)"} ?? ""
-        let path = req.parameters.getCatchall().joined(separator: "/") + anchor
+        let path = req.parameters.getCatchall().joined(separator: "/").lowercased() + anchor
 
         guard let target = try await DocumentationTarget.query(on: req.db,
                                                                owner: owner,

--- a/Tests/AppTests/PackageController+routesTests.swift
+++ b/Tests/AppTests/PackageController+routesTests.swift
@@ -276,6 +276,11 @@ class PackageController_routesTests: AppTestCase {
             XCTAssertEqual($0.status, .seeOther)
             XCTAssertEqual($0.headers.location, "/owner/package/1.0.0/documentation/foo#anchor")
         }
+        try app.test(.GET, "/owner/package/documentation/FOO") {
+            // Ensure we redirect to lowercase path URLs (which is what DocC generates.
+            XCTAssertEqual($0.status, .seeOther)
+            XCTAssertEqual($0.headers.location, "/owner/package/1.0.0/documentation/foo")
+        }
     }
 
     func test_documentationRoot_redirect() throws {

